memcached.tf

```diff
@@ -4,13 +4,28 @@ provider "aws" {
   region      = "${var.aws_region}"
 }
 
+resource "aws_elasticache_subnet_group" "demo_memcached_subnet_group" {
```

Comment:
what is this for? why do we need a subnet group?

Reply:
This resource is for helping Amazon Elasticache to identify the subnets ( in a VPC ) to be used for spinning up the cache nodes.

The resource [aws_elasticache_cluster](https://www.terraform.io/docs/providers/aws/r/elasticache_cluster.html) accepts only [aws_elasticache_subnet_group](https://www.terraform.io/docs/providers/aws/r/elasticache_subnet_group.html) and there is no argument like `subnet_id`. So, we are forced to create this resource, even if we need to mention only one subnet in our subnet group.

Reply:
why is this happening on public groups? then what was the point of the private group?

Reply:
Do you mean `vpc_public_sn_id` being used in this subnet group?

Our current implementation looks like this reference picture from [AWS docs](https://docs.aws.amazon.com/AmazonElastiCache/latest/mem-ug/VPCs.EC.html)
![](https://docs.aws.amazon.com/AmazonElastiCache/latest/mem-ug/images/vpc-overview-diagram.png)
Image Source: https://docs.aws.amazon.com/AmazonElastiCache/latest/mem-ug/VPCs.EC.html

Both the memcached and EC2 instances are spin up in the same subnet, but we use two different  security groups to make sure that EC2 instances are available to the outside world and memcached cluster is available only to the machines inside the subnet.